src/commands/deploy.ts

```diff
@@ -49,7 +52,29 @@ export async function deploy(telemetryProperties: { [key: string]: string; }, tr
 
     await verifyRuntimeIsCompatible(runtime, outputChannel, client, siteWrapper);
 
-    await siteWrapper.deploy(folderPath, client, outputChannel, extensionPrefix);
+    node.treeItem.state = localize('deploying', 'Deploying...');
+    try {
+        node.refresh();
+        await siteWrapper.deploy(folderPath, client, outputChannel, extensionPrefix);
+    } finally {
+        node.treeItem.state = await node.treeItem.siteWrapper.getState(client);
```

Comment:
Do you care that setting the state here could get an error, leaving it at deploying...?

Reply:
Hmm good point. I don't think I care for now. I created a new issue:
https://github.com/Microsoft/vscode-azurefunctions/issues/133

There's a larger problem where refresh should update the state anyways (aka I wouldn't have to update it here) and I'll fix it as a part of that